homeassistant/components/media_player/frontier_silicon.py

```diff
@@ -0,0 +1,261 @@
+    SUPPORT_VOLUME_STEP, SUPPORT_STOP, SUPPORT_TURN_OFF, SUPPORT_TURN_ON,
+    SUPPORT_PLAY, SUPPORT_SELECT_SOURCE, MediaPlayerDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (
+    STATE_OFF, STATE_PLAYING, STATE_PAUSED, STATE_UNKNOWN,
+    CONF_HOST, CONF_PORT, CONF_PASSWORD)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['fsapi==0.0.7']
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_FRONTIER_SILICON = SUPPORT_PAUSE | SUPPORT_VOLUME_SET | \
+    SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP | \
+    SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK | SUPPORT_SEEK | \
+    SUPPORT_PLAY_MEDIA | SUPPORT_PLAY | SUPPORT_STOP | SUPPORT_TURN_ON | \
+    SUPPORT_TURN_OFF | SUPPORT_SELECT_SOURCE
+
+DEFAULT_HOST = '192.168.1.11'
+DEFAULT_PORT = 80
+DEFAULT_PASSWORD = '1234'
+DEVICE_URL = 'http://{0}:{1}/device'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_PASSWORD, default=DEFAULT_PASSWORD): cv.string,
+})
+
+
+# pylint: disable=unused-argument
+# def setup_platform(hass, config, add_devices, discovery_info=None):
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Setup the Frontier Silicon platform."""
+    if discovery_info is not None:
+        yield from async_add_entities(
+            [FrontierSiliconDevice(discovery_info, DEFAULT_PASSWORD)],
+            update_before_add=True)
+        return True
+
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    password = config.get(CONF_PASSWORD)
+
+    try:
+        if host and port and password:
+            yield from async_add_entities(
+                [FrontierSiliconDevice(DEVICE_URL.format(host, port), password)],
+                update_before_add=True)
+            _LOGGER.info('FSAPI device %s:%s -> %s', host, port, password)
+            return True
+        else:
+            _LOGGER.warning('FSAPI device missing config param %s:%s -> %s',
+                            host, port, password)
+    except requests.exceptions.RequestException:
+        _LOGGER.error('Could not connect to FSAPI device at %s:%s -> %s',
+                      host, port, password)
+
+
+class FrontierSiliconDevice(MediaPlayerDevice):
+    """Representation of a Frontier Silicon device on the network."""
+
+    def __init__(self, device_url, password):
+        """Initialize the Frontier Silicon API device."""
+        self._device_url = device_url
+        self._password = password
+        self._state = STATE_UNKNOWN
+
+        self._name = None
+        self._title = None
+        self._mute = None
+        self._source = None
+        self._source_list = None
+        self._media_image_url = None
+
+    def get_fs(self):
+        """ Create a fsapi session."""
+        from fsapi import FSAPI
+        return FSAPI(self._device_url, self._password)
```

Comment:
afaik this will block the event loop as FSAPI includes a HTTP request
I guess you should either define get_fs() as a coroutine and yield the FSAPI call or leave get_fs() as it is, but yield it each time when it is called.